### PR TITLE
Update komoju gateway

### PR DESCRIPTION
**Pull request by shioyama**
This updates the Komoju gateway to be almost the same as what we use in [komoju/spree_komoju](https://github.com/komoju/spree_komoju).

### CODE REVIEW

**Comment by shioyama** [comment]
@k2nr Could you review this? I think you know the AM gateway code better than anyone else (although it's been a long time...)

File: lib/active_merchant/billing/gateways/komoju.rb [hunk 1]
```diff
@@ -51,50 +54,66 @@ def capture(*args)
...
       private
 
-      def payment_details(payment, options)
-        details = if payment.is_a?(CreditCard)
-                    credit_card_payment_details(payment, options)
-                  else
-                    payment
-                  end
-
-        details[:email] = options[:email] if options[:email] && details.is_a?(Hash)
```
Comment (shioyama) [inline lib/active_merchant/billing/gateways/komoju.rb:67]: Seems maybe one difference here is that `options[:email]` is not assigned to `details[:email]` unless details is a Hash. But we're always sending the payment as a hash, so this has no actual impact.

**Comment by k2nr** [comment]
Looks good to me :+1:

**Comment by shioyama** [comment]
Thanks!

**Pull request merged**